### PR TITLE
BCDA-7655: Update SSAS for resiliency / connection management improvements. 

### PR DESCRIPTION
**Pull request by alex-dzeda**
## 🎫 Ticket

https://jira.cms.gov/browse/BCDA-7655

## 🛠 Changes

Updated maximum number of DB connections from 40 -> 60
Added idle connection timeout
Updated from go 1.18-> 1.19 concurrent with updating to latest versions of dependencies.
Removed ioutil references (deprecation caused issues going from 1.18-> 1.19)

## ℹ️ Context for reviewers

There are inefficiencies in connection handling for the go postgres / sql libraries that we use, leading to issues with stability. While bcda-app is locked to an old version of pgx due to the queuing package used, there’s no such restriction for SSAS, so we can use the latest version of the packages (with connection pooling improvements). 

## ✅ Acceptance Validation

Manual verification of information in the commit, unit tests pass. 

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

**Comment by laurenkrugen-navapbc** [comment]
` This PR adds a new software dependency or dependencies.` - should we add a platform or sec team member to this PR since we are upgrading to 1.19?

**Comment by alex-dzeda** [comment]
> ` This PR adds a new software dependency or dependencies.` - should we add a platform or sec team member to this PR since we are upgrading to 1.19?

Sure thing, adding @gsf as well for a quick look-see. 
NB, the only mandated change with 1.18-> 1.19 was removal of ioutil references, which was simple enough.

File: ssas/connection.go [hunk 1]
```diff
@@ -31,9 +31,10 @@ func createDB() (*gorm.DB, error) {
 
 	// TODO: Allow connection settings to be configured by env vars
 	// https://jira.cms.gov/browse/BCDA-7109
-	sqlDB.SetMaxOpenConns(40)
+	sqlDB.SetMaxOpenConns(60)
```
Comment (kyeah) [inline ssas/connection.go:34]: For reference, I checked production and the database-level max connections is pretty high:

```
bcda=> show max_connections;
 max_connections 
-----------------
 1709
```

File: ssas/connection.go [hunk 2]
```diff
@@ -31,9 +31,10 @@ func createDB() (*gorm.DB, error) {
...
 	sqlDB.SetMaxIdleConns(40)
 	sqlDB.SetConnMaxLifetime(time.Duration(5) * time.Minute)
+	sqlDB.SetConnMaxIdleTime(time.Duration(30) * time.Second)
```
Comment (kyeah) [inline ssas/connection.go:37]: Didn't realize this was a method, thank you!

**Review by kyeah — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**